lib/types/meta_data.ex

```diff
@@ -0,0 +1,94 @@
+defmodule Kadena.Types.MetaData do
+  @moduledoc """
+  `MetaData` struct definition.
+  """
+
+  alias Kadena.Types.ChainID
+
+  @behaviour Kadena.Types.Spec
+
+  @type creation_time :: number()
+  @type ttl :: number()
+  @type gas_limit :: number()
+  @type gas_price :: number()
+  @type sender :: String.t()
+  @type chain_id :: ChainID.t()
+  @type chain_id_arg :: String.t()
+  @type values :: creation_time() | ttl() | gas_limit() | gas_price() | sender() | chain_id()
+  @type validation :: {:ok, values()} | {:error, Keyword.t()}
+
+  @type t :: %__MODULE__{
+          creation_time: creation_time(),
+          ttl: ttl(),
+          gas_limit: gas_limit(),
+          gas_price: gas_price(),
+          sender: sender(),
+          chain_id: chain_id()
+        }
+
+  defstruct [:creation_time, :ttl, :gas_limit, :gas_price, :sender, :chain_id]
+
+  @impl true
+  def new(args) do
+    creation_time = Keyword.get(args, :creation_time)
+    ttl = Keyword.get(args, :ttl)
+    gas_limit = Keyword.get(args, :gas_limit)
+    gas_price = Keyword.get(args, :gas_price)
+    sender = Keyword.get(args, :sender)
+    chain_id = Keyword.get(args, :chain_id)
+
+    with {:ok, creation_time} <- validate_creation_time(creation_time),
+         {:ok, ttl} <- validate_ttl(ttl),
+         {:ok, gas_limit} <- validate_gas_limit(gas_limit),
+         {:ok, gas_price} <- validate_gas_price(gas_price),
+         {:ok, sender} <- validate_sender(sender),
+         {:ok, chain_id} <- validate_chain_id(chain_id) do
+      %__MODULE__{
+        creation_time: creation_time,
+        ttl: ttl,
+        gas_limit: gas_limit,
+        gas_price: gas_price,
+        sender: sender,
+        chain_id: chain_id
+      }
+    end
+  end
+
+  @spec validate_creation_time(creation_time :: creation_time()) :: validation()
+  defp validate_creation_time(creation_time) when is_number(creation_time),
+    do: {:ok, creation_time}
+
+  defp validate_creation_time(_creation_time), do: {:error, [creation_time: :invalid]}
+
+  @spec validate_ttl(ttl :: ttl()) :: validation()
+  defp validate_ttl(ttl) when is_number(ttl),
```

Comment:
@CristhianRodriguezMolina @santiagoboterokommit Why not use a single function to validate numbers? This will allow you to group `validate_ttl` `validate_gas_limit` `validate_gas_price` into a single validation function. 

Take a look at `Kadena.Types.KeyPair#26`.